### ISSUE TITLE
Clang-Format removes space after preprocessor directive

### ISSUE THREAD

**Issue by RobertZickler** [llvm/llvm-project#117706]
Give the following test code:

```c
// Good test
#error My error message
#error - My error message
#error == My error message
#error # My error message
#error + My error message
// Bad test
#error -- My error message
#error ++ My error message
#error . My error message
#error ## My error message
#error (My error message)
#error ++My error message
//   ^^^
```
Clang-Format (v19.1.0) formats it like this:

```c
// Good test
#error My error message
#error - My error message
#error == My error message
#error # My error message
#error + My error message
// Bad test
#error-- My error message
#error++ My error message
#error. My error message
#error## My error message
#error(My error message)
#error++ My error message
//   ^^^
```

The space after the preprocessor directive is sometimes removed. That is leading to _dodgy_ code. This also happens with `#warning` and probably more directives.

This can be avoided by using quotes, but they are not required.

On a side note: This is leading to a MISRA-C:2012 required warning on the bad formatted lines (rule 20.13).

**Comment by owenca** [comment]
> Give the following test code:
> 
> // Good test
> #error My error message
> #error - My error message
> #error == My error message
> #error # My error message
> #error + My error message
> // Bad test
> #error -- My error message
> #error ++ My error message
> #error . My error message
> #error ## My error message
> #error (My error message)
> #error ++My error message
> //   ^^^

Isn't this invalid C code?

**Comment by RobertZickler** [comment]
Yes, you are right. `#error` will not compile due to being a forced error. I should have been precise. But replace it with `#warning` and it will compile. With and without the space after the `#warning`.

You need to compile **without** the options `-pedantic`, `-pedantic-errors`, and `-Werror`.

Given the warning

```c
#warning-- My warning message
```

without a space (due to Clang-Format), it is still building successfully and prints 

> warning: #warning -- My warning message [-Wcpp]

during the compilation. The space after the warning is added in the output.

Building with `gcc.exe (Rev3, Built by MSYS2 project) 14.1.0`
